SVCPLAN-4146 Add error detection to puppet_run postscript

bsper2 opened the pull request.

bsper2 committed:
SVCPLAN-4146 Add error detection to puppet_run postscript

jakerundall reviewed (approved):
This looks good. My only inputs are:

(1) If we retry and fail until we hit retries # of runs, then we croak, then we skip starting/enabling Puppet. I'm not sure if that's the right thing or not. Thoughts? If there were a problem that's eventually resolved (e.g., node is set to a non-existent environment, but then that's fixed later) we might want the agent to be started/enabled so things would eventually resolve?

(2) I'd add a comment before the check for Puppet exit codes 0 or 2 just to explain what those mean / what outcomes we're checking for.

bsper2 commented:
(1) True, enabling puppet does sounds like the right thing to do even if the agent exceeds the retry count. Added that in so it will enable puppet before it croaks.

(2) Sounds good, added a comment that details what those two exit codes mean.
